### PR TITLE
LibWeb: Implement HTMLFormControlsCollection

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/RadioNodeList.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    return m_node_list.size();
+}
+
+// https://dom.spec.whatwg.org/#dom-nodelist-item
+Node const* RadioNodeList::item(u32 index) const
+{
+    // The item(index) method must return the indexth node in the collection. If there is no indexth node in the collection, then the method must return null.
+    if (index >= m_node_list.size())
+        return nullptr;
+    return static_cast<ParentNode const*>(m_node_list[index].ptr());
```
Comment (shannonbooth) [inline Userland/Libraries/LibWeb/DOM/RadioNodeList.cpp:46]: This `static_cast` doesn't really feel correct here, I feel like I may be using wrong approach for this?

File: Userland/Libraries/LibWeb/DOM/RadioNodeList.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    virtual u32 length() const override;
+    virtual Node const* item(u32) const override;
+
+protected:
+    virtual void initialize(JS::Realm&) override;
+
+private:
+    explicit RadioNodeList(JS::Realm&, Vector<JS::Handle<Element>> const&);
+
+    Vector<JS::Handle<Element>> m_node_list;
```
Comment (shannonbooth) [inline Userland/Libraries/LibWeb/DOM/RadioNodeList.h:31]: Not 100% on whether this should be an `Element` or not. Also using a Handle right now, but maybe this should be GCCounted? Really unsure of the differences between all that

Comment (awesomekling) [reply Userland/Libraries/LibWeb/DOM/RadioNodeList.h:31]: As I understand it, `RadioNodeList` is supposed to be a "live" collection, meaning that if radio elements are removed from the DOM, they should no longer be present in the `RadioNodeList` either. So I think we shouldn't be storing them here like this in the first place :)

Comment (shannonbooth) [reply Userland/Libraries/LibWeb/DOM/RadioNodeList.h:31]: Right, that would make a lot more sense, and explains my difficulty! Glad that doing the wrong thing was difficult :P

Will give this another shot, thanks!

Comment (shannonbooth) [reply Userland/Libraries/LibWeb/DOM/RadioNodeList.h]: Thanks, this looks much better now!

**Pull request merged**